genesyscloud/user_group_role_common.go

```diff
@@ -247,3 +247,64 @@ func validateResourceRole(resourceName string, roleResourceName string, division
 		return fmt.Errorf("Missing expected role for resource %s in state: %s", resourceID, roleID)
 	}
 }
+func fetchRoleIds(ctx context.Context, d *schema.ResourceData, meta interface{}, roles *schema.Set) *resource.RetryError {
+
+	sdkConfig := meta.(*ProviderMeta).ClientConfig
+	usersApi := platformclientv2.NewUsersApiWithConfig(sdkConfig)
+	domainRoleIds, _, err := getDomainIdRoles(d.Id(), usersApi)
+
+	if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("%v", err))
+	}
+
+	roleSlice := roles.List()
+	missingroleIDs := []string{}
+
+	for _, id := range domainRoleIds {
+			// search for the id in the set of maps
+			found := false
+			for _, roleElem := range roleSlice {
+					role := roleElem.(map[string]interface{})
+					if role["role_id"].(string) == id {
+							found = true
+							break
+					}
+			}
+			if !found {
+					missingroleIDs = append(missingroleIDs, id)
+			}
+	}
+
+	if len(missingroleIDs) > 0 {
+			return resource.RetryableError(fmt.Errorf("The following roles (Ids) are not attached for user id %s: %s", d.Id(), missingroleIDs))
+	}
+
+	return nil
+}
+
+func getDomainIdRoles(subjectID string, userAPI *platformclientv2.UsersApi) ([]string, *platformclientv2.APIResponse, diag.Diagnostics) {
+	var roles []string
+
+	data, response, err := userAPI.GetUserRoles(subjectID)
+	if err != nil {
+			return nil, response, diag.Errorf("Failed to get Roles for subject %s: %s", subjectID, err)
+	}
+
+	if data != nil && data.Roles != nil {
+			for _, role := range *data.Roles {
+					if role.Id != nil {
```

Comment:
roleId will never be null here.  Also, you should never have a data.Roles as nil.  It is as defined as an array and the SDK will translate the data as an empty array